### PR TITLE
dm/dmm code cleanup and fix dmj json output

### PR DESCRIPTION
**Pull request by daddesio**
* Always escape filename/path when printing json. (Fixes dmj output
  on Windows.)
* Always wrap path in quotes when printing r2 commands (even on
  non-Windows).
* Assume memory pages belonging to modules always have map->path and
  map->name set (i.e. remove redundant NULL comparisons).
* Assume memory pages _in general_ do not (per commit a1e24700e3c90182).
* In dm command, compare flag name against r_name_filter(map->name).
* Fix dm output for user maps.

On Windows, dmj's "name" (map->name) field may contain a backslash (e.g. "MAPPED   \DeviceHarddiskVolume2\Windows\System32\locale.nls") which needs to be escaped.

Note: This is my first patch for radare2, so if you don't like my coding style, just mention it and I'll fix it right away, thanks.

### CODE REVIEW

File: libr/core/cmd_debug.c [hunk 1]
```diff
@@ -1331,85 +1331,37 @@ static void cmd_debug_modules(RCore *core, int mode) { // "dmm"
...
 			if (addr >= map->addr && addr < map->addr_end) {
 				r_cons_printf ("0x%08"PFMT64x" %s\n", map->addr, map->file);
 				goto beach;
 			}
 			break;
 		case 'j':
-#if __WINDOWS__ && !__CYGWIN__
 			{
-				/* Single backslashes cause issues when parsing JSON output, so escape them */
+				/* Escape backslashes (e.g. for Windows). */
```
Comment (XVilka) [inline libr/core/cmd_debug.c]: Why did you remove `__WINDOWS__ && !__CYGWIN__`? Escaping backslashes is not required for non-Windows system.

Comment (radare) [reply libr/core/cmd_debug.c]: It shouldnt hurt

File: libr/debug/map.c [hunk 2]
```diff
@@ -4,124 +4,102 @@
...
+				if (map->file && *map->file) {
+					char *escaped_path = r_str_escape (map->file);
+					dbg->cb_printf ("\"file\":\"%s\",", escaped_path);
+					free (escaped_path);
+				}
+				dbg->cb_printf ("\"addr\":%"PFMT64u",", map->addr);
+				dbg->cb_printf ("\"addr_end\":%"PFMT64u",", map->addr_end);
+				dbg->cb_printf ("\"type\":\"%c\",", map->user?'u':'s');
+				dbg->cb_printf ("\"perm\":\"%s\"", r_str_rwx_i (map->perm));
+				dbg->cb_printf ("}%s", iter->n? ",": "");
```
Comment (daddesio) [inline libr/debug/map.c]: Oops, the "if iter->n > 0, then print comma" check won't work anymore since we're iterating over two lists.

**Comment by radare** [comment]
Send another pr

File: libr/core/cmd_debug.c (hunk 1 above)
Comment (XVilka) [reply libr/core/cmd_debug.c]: Why? It hurts.

**Comment by radare** [comment]
Its a string in json format. Theres only one way to encode it. No matter if its windows or linux
